### PR TITLE
MPV module updated

### PR DESCRIPTION
**Pull request by yeonjaej**
Low E kinematic coverage is modified in the module. 
The change can be reverted by setting `physics.producers.generator.Revert` to 1 in fcl config.
`gen_mpvmpr.fcl` now includes single gamma in MPV. 2-7 particles in an interaction are generated for MPV.
MPV particle multiplicity and species are configurable in the fcl.
Boost implementation (Work By Joshua Berger) can be turned on `physics.producers.generator.UseBoost` to 1.
Boost range can be configured by setting `physics.producers.generator.GammaBetaRange`

### CODE REVIEW

File: sbncode/EventGenerator/MultiPart/MultiPartVertex_module.cc [hunk 1]
```diff
@@ -37,155 +37,168 @@
...
+  for(size_t idx=0; idx<minmult_v.size(); ++idx) {
+    if(minmult_v[idx] > maxmult_v[idx]) this->abort("Particle MinMulti > Particle MaxMulti!");
+    if(minmult_v[idx] > _multi_max) this->abort("Particle MinMulti > overall MultiMax!");
+    multi_min += minmult_v[idx];
+  }
+  _multi_min = std::max(_multi_min, multi_min);
+  if(_multi_max < _multi_min) this->abort("Overall MultiMax <= overall MultiMin!");
 
-    /*
+  /*
```
Comment (bear-is-asleep) [inline sbncode/EventGenerator/MultiPart/MultiPartVertex_module.cc:201]: should we just delete these lines?

File: sbncode/EventGenerator/MultiPart/MultiPartVertex_module.cc [hunk 2]
```diff
@@ -197,182 +210,209 @@ MultiPartVertex::MultiPartVertex(fhicl::ParameterSet const & p)
...
+
+void MultiPartVertex::GenBoost(double& bx, double& by, double& bz) {
+
+  double gbmag = 0.;
+  if ( _gamma_beta_range[1] > _gamma_beta_range[0]) {
+    gbmag = fFlatRandom->fire(_gamma_beta_range[0],_gamma_beta_range[1]);
+  }
+  else {
+    gbmag = _gamma_beta_range[0];
+  }
```
Comment (bear-is-asleep) [inline sbncode/EventGenerator/MultiPart/MultiPartVertex_module.cc:398]: Should these checks be done when first reading the cfg? Also for the else, do we just want to select the lowest one or should we assert the second element is larger than the first?

Comment (bear-is-asleep) [reply sbncode/EventGenerator/MultiPart/MultiPartVertex_module.cc:398]: i see that other places default to the first element so i bet this is fine.

File: sbncode/EventGenerator/MultiPart/MultiPartVertex_module.cc (hunk 1 above)
Comment (bear-is-asleep) [reply sbncode/EventGenerator/MultiPart/MultiPartVertex_module.cc:201]: we can probably decide to delete these one day, in a future PR

File: sbncode/EventGenerator/MultiPart/MultiPartVertex_module.cc [hunk 3]
```diff
@@ -37,155 +37,168 @@
...
+  _multi_min = p.get<size_t>("MultiMin");
+  _multi_max = p.get<size_t>("MultiMax");
+
+  _tpc_v  = p.get<std::vector<std::vector<unsigned short> > >("TPCRange");
+  auto const xrange = p.get<std::vector<double> > ("XRange");
+  auto const yrange = p.get<std::vector<double> > ("YRange");
+  auto const zrange = p.get<std::vector<double> > ("ZRange");
+
+  auto const part_cfg = p.get<fhicl::ParameterSet>("ParticleParameter");
+  _gamma_beta_range = p.get<std::array<double,2> >("GammaBetaRange", {0.0,0.0});
```
Comment (bear-is-asleep) [inline sbncode/EventGenerator/MultiPart/MultiPartVertex_module.cc:159]: im confused if this stores velocity or gamma

File: sbncode/EventGenerator/MultiPart/MultiPartVertex_module.cc [hunk 4]
```diff
@@ -37,155 +37,168 @@
...
+  if( pdg_v.size() != kerange_v.size() ||
+      pdg_v.size() != minmult_v.size() ||
+      pdg_v.size() != maxmult_v.size() ||
+      pdg_v.size() != weight_v.size() )
+    this->abort("configuration parameters have incompatible lengths!");
+
+  // further sanity check (1 more depth for some double-array)
+  for(auto const& r : pdg_v    ) { if(              r.empty()  ) this->abort("PDG code not given!");                        }
+  for(auto const& r : kerange_v) { if(              r.size()!=2) this->abort("Incompatible legnth @ KE vector!");           }
+  if(_gamma_beta_range[0] > _gamma_beta_range[1]) this->abort("Incompatible boost range!");
```
Comment (bear-is-asleep) [inline sbncode/EventGenerator/MultiPart/MultiPartVertex_module.cc:190]: if this is velocity, we should check if both are bounded between 0 and 1

Comment (yeonjaej) [reply sbncode/EventGenerator/MultiPart/MultiPartVertex_module.cc:190]: `_gamma_beta` denotes gamma times beta, so the possible range is 0 to inf.

File: sbncode/EventGenerator/MultiPart/MultiPartVertex_module.cc (hunk 3 above)
Comment (yeonjaej) [reply sbncode/EventGenerator/MultiPart/MultiPartVertex_module.cc:159]: beta_gamma stores beta*gamma

**Review by bear-is-asleep — APPROVED** [review]
I left prior comments that are minor, but other than that this looks good!

**Pull request merged**